sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -1547,14 +1547,25 @@ case class BRound(child: Expression, scale: Expression)
 }
 
 object WidthBucket {
-
   def computeBucketNumber(value: Double, min: Double, max: Double, numBucket: Long): jl.Long = {
-    if (numBucket <= 0 || numBucket == Long.MaxValue || jl.Double.isNaN(value) || min == max ||
-        jl.Double.isNaN(min) || jl.Double.isInfinite(min) ||
-        jl.Double.isNaN(max) || jl.Double.isInfinite(max)) {
-      return null
+    if (isNull(value, min, max, numBucket)) {
+      null
+    } else {
+      computeBucketNumberNotNull(value, min, max, numBucket)
     }
+  }
 
+  def isNull(value: Double, min: Double, max: Double, numBucket: Long): Boolean = {
```

Comment:
nit: this method does not need to be public, can be `private def`.

Reply:
Are you sure? The Java code generated for `WholeStageCodegenExec` won't be part of the `WidthBucket` class, so I don't think it can call `WidthBucket.isNull` if the function is `private`.

Reply:
Sorry yeah you are right. Shall we add a comment for both `isNull` and `computeBucketNumberNotNull` to explain it?

```
/**
 * This is called by generated Java class, should be public.
 */
```
Similar to [`HashAggregateExec`](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala#L426-L429).

Reply:
Good idea! I will add it.